metaflow/plugins/tagging_cli.py

```diff
@@ -0,0 +1,303 @@
+@click.option('--namespace',
+              'user_namespace',
+              required=False,
+              default=None,
+              type=str,
+              help="Change namespace from the default (your username) to "
+                   "the specified tag.")
+@click.argument('tags',
+                required=True,
+                type=str,
+                nargs=2)
+@click.pass_obj
+def replace(obj, run_id, user_namespace, tags):
+    user_namespace = resolve_identity() if user_namespace is None else user_namespace
+    metaflow_run_client = obtain_metaflow_run_client(obj,
+                                                     run_id, user_namespace)
+
+    metaflow_run_client.replace_tag([tags[0]], [tags[1]])
+
+    obj.echo("Operation successful. New tags:")
+    obj.echo(print_tags(metaflow_run_client))
+
+@tag.command('list',
+             help='List tags of a run.')
+@click.option('--run-id',
+              required=False,
+              default=None,
+              type=str,
+              help="Run ID of the specific run to list.")
+@click.option('--all',
+              'list_all',
+              required=False,
+              is_flag=True,
+              default=False,
+              help="List tags across all runs of this flow.")
+@click.option('--my-runs',
+              'my_runs',
+              required=False,
+              is_flag=True,
+              default=False,
+              help="List tags across all runs of the flow under the default namespaces.")
+@click.option('--hide-system-tags',
+              required=False,
+              is_flag=True,
+              default=False,
+              help="Hide system tags.")
+@click.pass_obj
+def tag_list(obj, run_id, hide_system_tags, list_all, my_runs):
+    if (run_id is None and not list_all and not my_runs):
+        raise CommandException(
+            "Please use one of the options --run-id / --all / --my-runs.\n" +
+            "You can run the flow with commands 'tag list --help' " +
+            "to check the meaning of the three options."
+        )
+
+    if (list_all and my_runs):
+        raise CommandException("Option --all cannot be used together with --my-runs.\n" +
+                               "You can run the flow with commands 'tag list --help' " +
+                               "to check the meaning of the two options.")
+
+    system_tags = set()
+    all_tags = set()
+
+    if list_all or my_runs:
+        user_namespace = resolve_identity() if my_runs else None
+        runs = list_runs(obj, user_namespace)
+        for run in runs:
+            system_tags.update(run.system_tags)
+            all_tags.update(run.tags)
+    else:
+        metaflow_run_client = obtain_metaflow_run_client(obj, run_id, None)
+        system_tags = metaflow_run_client.system_tags
+        all_tags = metaflow_run_client.tags
+
+    if hide_system_tags:
+        obj.echo(format_tags([], all_tags.difference(system_tags)), err=False)
```

Comment:
we should make `list` respect the `--quiet` flag, i.e. use `obj.echo_always` for data and `obj.echo` for informational lines. Compare this to `--quiet logs` vs. `logs`.

The point of `--quiet` is that it makes the output script-friendly (safe to pipe, `grep` etc)